### ISSUE TITLE
Unable to run a single end-to-end test

### ISSUE THREAD

**Issue by rhmdnd** [ComplianceAsCode/compliance-operator#226]
The `Makefile` documents a way to invoke a single end-to-end test. This is desirable since end-to-end tests take a while and being able iterate on a single test quickly significantly improves the feedback loop for developers.

However, the test runner ignores arguments I pass to invoke a single test (`--debug=j` for additional verbosity through `make`).

```
 $ E2E_GO_TEST_FLAGS="-v -run TestE2E/TestProfileModification" make --debug=j e2e
...
    client.go:68: resource type ServiceAccount with namespace/name (osdk-e2e-f512c609-3e9d-4866-b6e8-845ed4008cfa/rerunner) successfully deleted
    client.go:68: resource type ServiceAccount with namespace/name (osdk-e2e-f512c609-3e9d-4866-b6e8-845ed4008cfa/remediation-aggregator) successfully deleted
    client.go:68: resource type ServiceAccount with namespace/name (osdk-e2e-f512c609-3e9d-4866-b6e8-845ed4008cfa/profileparser) successfully deleted
    client.go:68: resource type ServiceAccount with namespace/name (osdk-e2e-f512c609-3e9d-4866-b6e8-845ed4008cfa/compliance-operator) successfully deleted
    client.go:68: resource type ServiceAccount with namespace/name (osdk-e2e-f512c609-3e9d-4866-b6e8-845ed4008cfa/api-resource-collector) successfully deleted
--- PASS: TestE2E (205.55s)
testing: warning: no tests to run
PASS
ok      github.com/ComplianceAsCode/compliance-operator/tests/e2e       208.005s [no tests to run]
Reaping winning child 0x5563c1c6d560 PID 783792
Removing child 0x5563c1c6d560 PID 783792 from chain.
```

I also noticed I'm unable to list tests:

```
go test ./tests/e2e -list=TestE2E -args $(E2E_ARGS)
```

**Comment by rhmdnd** [comment]
Seems like test discovery isn't working the way we expect.

**Comment by rhmdnd** [comment]
I'm fairly certain we're experiencing this issue because CO e2e tests are nested. The main test runner is implemented inside a test (`TestE2E`), where it organizes the test run to ensure all non-destructive tests are run concurrently and before serial tests.

The serial tests perform things like remediations, which could impact other tests, hence their need to be run serially and last.

The `TestE2E` test invokes all other tests, so when we use `-run` we don't get a complete match since `t.Run()` is called on `TestE2E` first. A mismatch here causes the entire suite to get skipped (e.g., `testing: warning: no tests to run`). You get a match if you use `-run TestE2E`, but that defeats the purpose since it's the equivalent to running the entire suite anyway.

Two options I can think of right now are to implement our own regular expression filter (implemented in PR #227), or to reorganize the tests to be flat so we can rely on the native `-run` implementation.

It's also important to note that `TestE2E` is implemented to setup all cluster resources once and share them across sub-test runs. This saves time since we're only installing the operator once instead of for each test (which I suspect would drastically increase test execution times).

If we can reorganize a flat test structure that shares test utilities, we could improve developer ergonomics without sacrificing test times.

**Comment by rhmdnd** [comment]
My hypothesis was wrong and while there are part of the test code that could be improved to leverage newer features of go (t.Cleanup() and testing.M), it was simply because I based my example on the documentation in the Makefile...

Using `TestE2E/Parallel_tests/TestScanWithNodeSelectorFiltersCorrectly` works fine.

Adding a patch to make that more clear.
